platforms/common/include/px4_platform_common/crypto.h

```diff
@@ -0,0 +1,211 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2021 Technology Innovation Institute. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#pragma once
+
+#ifdef PX4_CRYPTO
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <px4_random.h>
+#include <px4_platform_common/crypto_algorithms.h>
+#include <px4_platform_common/sem.h>
+#include "crypto_backend_definitions.h"
+
+/*
+ * Crypto API interface class
+ */
+
+class PX4Crypto
+{
+public:
+
+	/*
+	 * Crypto api initialization, called only once at boot
+	 */
+
+	static void px4_crypto_init(void);
+
+	/*
+	 * Class function to open and return a pointer to crypto api
+	 */
+
+	static PX4Crypto *px4_crypto_open(px4_crypto_algorithm_t algorithm);
```

Comment:
I'd rather have a public constructor and then `bool open(px4_crypto_algorithm_t algorithm);`. That way you can avoid the extra dynamic allocation and forget about the close call (done in the destructor), or at least make it optional.